packages/jest-jasmine2/src/index.js

```diff
@@ -69,7 +69,32 @@ function jasmine2(config, environment, moduleLoader, testPath) {
       moduleLoader.requireModule(null, config.setupTestFrameworkScriptFile);
     }
   });
+
+  const iterableEquality = (left, right) => {
+    let result; // result needs to be `undefined` if not an iterator
+    if (left && right && left[Symbol.iterator] && right[Symbol.iterator]) {
+      const leftIterator = left[Symbol.iterator]();
+      const rightIterator = right[Symbol.iterator]();
+      let nextLeft = leftIterator.next();
+      let nextRight = rightIterator.next();
+
+      result = true;
+      while (result && !nextLeft.done) {
+        result = jasmine.matchersUtil.equals(nextLeft, nextRight);
```

Comment:
I need to check both done and value otherwise the second iterator could end before the first, and if  first ends with any `undefined` it'd look the same:

`['a', undefined, undefined]` vs `['a']`